lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -1521,7 +1533,8 @@ void DeclAndTypeClangFunctionPrinter::printCxxMethod(
   auto result = printFunctionSignature(
       FD, signature, cxx_translation::getNameForCxx(FD), resultTy,
       FunctionSignatureKind::CxxInlineThunk, modifiers);
-  assert(!result.isUnsupported() && "C signature should be unsupported too");
```

Comment:
There are multiple code paths in `printFunctionSignature` that already return `unsupported`, even before this PR. I think we should handle that scenario gracefully here.